docs/howtos/extracting_intermediates.rst

```diff
@@ -0,0 +1,204 @@
+Extracting intermediate values
```

Comment:
What do you think of adding something like a TL;DR at the top with the three suggested approaches?